src/users/UserEdit.tsx

```diff
@@ -38,6 +40,7 @@ export default function UserEdit() {
       ...user,
       id: userId,
     });
+    navigate(-1);
```

Comment:
I would only call that in the `onSuccess` and it reminds me you haven't handled errors ;-)